finagle-core/src/test/scala/com/twitter/finagle/InetResolverTest.scala

```diff
@@ -76,4 +78,36 @@ class InetResolverTest extends FunSuite {
     assert(statsReceiver.stat("lookup_ms")().size > 0)
   }
 
+  test("updates using custom future pool") {
+    class TestPool(latch: CountDownLatch) extends FuturePool {
+      private val delegate = FuturePool.unboundedPool
+      override def apply[T](f: => T): Future[T] = {
+        latch.countDown()
+        delegate(f)
+      }
+    }
+
+    val latch = new CountDownLatch(1) // DnsResolver will use the pool
+    val resolvePool = new TestPool(latch)
+
+    val dnsResolverWithPool = new DnsResolver(DefaultStatsReceiver, resolvePool)
+
+    def resolveLoopback(host: String): Future[Seq[InetAddress]] = {
+      if (host.equals("127.0.0.1")) dnsResolverWithPool(host)
+      else Future.exception(new UnknownHostException())
+    }
+
+    val pollInterval = Duration(100, TimeUnit.MILLISECONDS)
```

Comment:
The timing here is a bit tight and might introduce some flakyness in this test case. Have you seen our `Time` api that allows you to manipulate the current time? See `Time.withCurrentTimeFrozen` etc. It's what we commonly use in finagle tests to introduce some determanistic behavior for things that deal in `Time` and seems like it should be useful here.